src/Blue/Tools/Api/Client.php

```diff
@@ -156,20 +156,29 @@ private function resolve(ResponseInterface $response)
 
             $attempts = $this->deferredResultMaxAttempts;
 
+            $locationHeader = $response->getHeader('location');
+
             while ($attempts > 0) {
+
+                if($locationHeader) {
```

Comment:
you mean a change to the existing public api?  what would the location header point to?